crates/pixi_manifest/src/toml/task.rs

```diff
@@ -1,16 +1,38 @@
-use pixi_toml::{OneOrMany, TomlFromStr, TomlIndexMap, TomlWith};
+use pixi_toml::{TomlFromStr, TomlIndexMap};
 use toml_span::{
     de_helpers::{expected, TableHelper},
     value::ValueInner,
     DeserError, Value,
 };
 
 use crate::{
-    task::{Alias, CmdArgs, Execute},
+    task::{Alias, CmdArgs, Dependency, Execute, TaskArg},
     warning::Deprecation,
     Task, TaskName, WithWarnings,
 };
 
+impl<'de> toml_span::Deserialize<'de> for TaskArg {
+    fn deserialize(value: &mut Value<'de>) -> Result<Self, DeserError> {
+        let mut th = match value.take() {
+            ValueInner::String(str) => {
+                return Ok(TaskArg {
+                    name: str.into_owned(),
+                    default: None,
+                })
+            }
+            ValueInner::Table(table) => TableHelper::from((table, value.span)),
+            inner => return Err(expected("string or table", inner, value.span).into()),
+        };
+
+        let name = th.required::<String>("name")?;
```

Comment:
```suggestion
        let name = th.required::<String>("arg")?;
```

@ruben-arts and I discussed naming, and we decided to call that key "arg" instead of "name" for consistency with "task". I also adapted the gist accordingly.